python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -2660,12 +2660,20 @@ def query_GPU_memory(note=""):
     cudaDF = cudaDF[boolmask]
 
     assert (
-        cudaDF.index._values.data_array_view.device_ctypes_pointer
-        == cudaDF["col0"].index._values.data_array_view.device_ctypes_pointer
+        cudaDF.index._values.data_array_view(
+            mode="write"
+        ).device_ctypes_pointer
+        == cudaDF["col0"]
+        .index._values.data_array_view(mode="write")
+        .device_ctypes_pointer
```

Comment:
I know it's only tests, but do we need to provoke copies here?

Reply:
Nope, not needed. Updated the test.